### DISCUSSION TITLE
How can i read the 200 ok response message first, then send the ack when INVITE

### DISCUSSION THREAD

**Discussion by ChuLiqiang** [resiprocate/resiprocate#292] [Q&A] [answered]
I want INVATE(S) ->200 OK(C) -> ACK(S)
(1)read some value from 200 ok response.for example A value.
(2)calculate B value by A value.put B value in ACK。
I find in ClientInviteSession.cxx. resip send ack first.then call callback to handler. have other interface to callback before send ack?

![image](https://user-images.githubusercontent.com/2747722/207779024-c59a9a9b-f0ca-4634-9a21-50a29198cbc8.png)

**Comment by sgodin** [comment, marked as answer]
Check out the info here:  
https://github.com/resiprocate/resiprocate/wiki/Modifying-Decorating-messages-sent

Scott

**Reply by ChuLiqiang** [reply]
I have read decorating-messages-sent it tell me how to modify message when send to other endpoint.I want change order for read 200 ok message  before send ack message. I can not find a way to modify this order.do you have a another way to change this order.I want not to modify resip code.ClientInviteSession is a source code in resip source code.

**Reply by sgodin** [reply]
Unfortunately, I don't see a way to accomplish what you want without modifying resip code.

**Reply by ChuLiqiang** [reply]
> Unfortunately, I don't see a way to accomplish what you want without modifying resip code.

Thank you very much.I will modify resip code to support this feature.
